pkg/collector/corechecks/containers/kubelet/provider/prometheus/provider.go

```diff
@@ -0,0 +1,197 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubelet
+
+package prometheus
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/prometheus/common/expfmt"
+	"github.com/prometheus/common/model"
+	"golang.org/x/exp/maps"
+
+	"github.com/DataDog/datadog-agent/pkg/aggregator/sender"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/containers/kubelet/common"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/kubelet"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// Provider provides the metrics related to data collected from a prometheus Kubelet endpoint.
+//
+// It is based on the python openmetrics mixin which is defined in the integrations-core repo, however in its current
+// form it has been trimmed down to focus on the functionality used by the kubelet check specifically. Do not assume that
+// this is a complete implementation of that.
+type Provider struct {
+	Config              *common.KubeletConfig
+	ScraperConfig       *ScraperConfig
+	transformers        map[string]func(*model.Sample, sender.Sender)
+	metricMapping       map[string]string
+	wildcardRegex       *regexp.Regexp
+	ignoredMetrics      map[string]bool
+	ignoredMetricsRegex *regexp.Regexp
+}
+
+type ScraperConfig struct {
+	Path string
+	// AllowNotFound determines whether the check should error out or just return nothing when a 404 status code is encountered
+	AllowNotFound bool
+}
+
+func NewProvider(config *common.KubeletConfig, transformers map[string]func(*model.Sample, sender.Sender), scraperConfig *ScraperConfig) *Provider {
```

Comment:
Since NewProvider is used in another package, it can be good to also define what is a Tranformer function signature

```suggestion
type TransformerFuncType func(*model.Sample, sender.Sender)
type TranformersType map[string] TransformerFuncType

func NewProvider(config *common.KubeletConfig, transformers map[string]func(*model.Sample, sender.Sender), scraperConfig *ScraperConfig) *Provider {
```

Reply:
Good callout, I had thought about doing it but I wasn't sure if it would be overkill or not. Your comment sort of sealed the deal with it! I have added the change